src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.cs

```diff
@@ -179,7 +192,7 @@ public static AnalyzerConfig Parse(SourceText text, string pathToFile)
             // Add the last section
             addNewSection();
 
-            return new AnalyzerConfig(globalSection!, namedSectionBuilder.ToImmutable(), pathToFile);
+            return new AnalyzerConfig(globalSection!, namedSectionBuilder.ToImmutable(), pathToFile, isGlobal: false);
```

Comment:
Why is this hard coded to `false` here? Given that any editor config can specify via property that it can be global I would expect this to be coming from a variable.

Reply:
So in the current design AnalyzerConfigs always parse as before, because you might have multiple configs with `isGlobal = true`. We then filter them and combine them to create a single global config. Would it make more sense if this property was called `isMergedlGlobal` or somesuch?

---
In reply to: [421675114](https://github.com/dotnet/roslyn/pull/43889#discussion_r421675114) [](ancestors = 421675114)

Reply:
This is not how I intended the type to be represented AnalyzerConfig should represent a fully-parsed analyzer config, independent of every other file. If you want to look at a group of analyzer configs, that's what AnalyzerConfigSet is for

Reply:
Renamed to `isMergedGlobal`

---
In reply to: [421696196](https://github.com/dotnet/roslyn/pull/43889#discussion_r421696196) [](ancestors = 421696196,421675114)

Reply:
And it is: it's still a fully parsed analyzer config, it just has a global property called is_global = true. When looked at as a group, we merge anything with is_global = true together to create a single 'merged' global config.

---
In reply to: [421790334](https://github.com/dotnet/roslyn/pull/43889#discussion_r421790334) [](ancestors = 421790334)